### PR TITLE
User attribute

### PR DESCRIPTION
**Pull request by HiradEmami**
Updates for User attribute and Partnership table

### CODE REVIEW

File: rest-api/src/main/java/org/madb/api/controller/BudgetController.java [hunk 1]
```diff
@@ -70,6 +70,7 @@ private void copyBudget(Budget from, Budget to) {
 		// never copy project ID
 		to.setBudget(from.getBudget());
 		to.setFinancialYear(from.getFinancialYear());
+                to.setUser(from.getUser());
```
Comment (raftAtGit) [inline rest-api/src/main/java/org/madb/api/controller/BudgetController.java]: Change your indentation settings in your IDE please, wrong indentation is everywhere

File: rest-api/src/main/java/org/madb/api/controller/PartnershipController.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+	
+	@DeleteMapping("/partnerships/{id}")
+	void deletePartnerships(@PathVariable Integer id) {
+		partnershipRepository.deleteById(id);
+	}
+	
+	private void copyPartnerships(Partnerships from, Partnerships to) {
+		// never copy project ID
+		
+                to.setUser(from.getUser());
```
Comment (raftAtGit) [inline rest-api/src/main/java/org/madb/api/controller/PartnershipController.java]: why not copying other fields?

and user should not be copies I guess, as it's the user who created the record. we dont want it to be changed

File: rest-api/src/main/java/org/madb/api/controller/ProjectController.java [hunk 3]
```diff
@@ -65,6 +65,7 @@ private void copyProject(Project from, Project to) {
 		to.setBudget(from.getBudget());
 		to.setDescription(from.getDescription());
 		to.setObservations(from.getObservations());
+                to.setUser(from.getUser());
```
Comment (raftAtGit) [inline rest-api/src/main/java/org/madb/api/controller/ProjectController.java]: user should not be copies I guess, as it's the user who created the record. we dont want it to be changed.
applied to other controllers too

File: rest-api/src/main/java/org/madb/api/model/Beneficiaries.java [hunk 4]
```diff
@@ -53,4 +53,7 @@
     @Column(name = "NUMBER_OF_BENEFICIARIES")
     private Integer number_of_beneficiaries;
     
+    @Size(max = 500)
```
Comment (raftAtGit) [inline rest-api/src/main/java/org/madb/api/model/Beneficiaries.java]: 500 characters  for user name? 50 will be more than enough

File: rest-api/src/main/java/org/madb/api/model/Beneficiaries.java [hunk 5]
```diff
@@ -53,4 +53,7 @@
...
+    @Column(name = "USER")
+    private String user;
```
Comment (raftAtGit) [inline rest-api/src/main/java/org/madb/api/model/Beneficiaries.java]: we dont want this to be empty, right? there should be a @NotNull constraint

File: rest-api/src/main/java/org/madb/api/controller/BudgetController.java (hunk 1 above)
Comment (HiradEmami) [reply rest-api/src/main/java/org/madb/api/controller/BudgetController.java]: Removed them.

File: rest-api/src/main/java/org/madb/api/controller/PartnershipController.java (hunk 2 above)
Comment (HiradEmami) [reply rest-api/src/main/java/org/madb/api/controller/PartnershipController.java]: Added them.

File: rest-api/src/main/java/org/madb/api/controller/ProjectController.java (hunk 3 above)
Comment (HiradEmami) [reply rest-api/src/main/java/org/madb/api/controller/ProjectController.java]: Removed them.

File: rest-api/src/main/java/org/madb/api/model/Beneficiaries.java (hunk 4 above)
Comment (HiradEmami) [reply rest-api/src/main/java/org/madb/api/model/Beneficiaries.java]: Adjusted them.

File: rest-api/src/main/java/org/madb/api/model/Beneficiaries.java (hunk 5 above)
Comment (HiradEmami) [reply rest-api/src/main/java/org/madb/api/model/Beneficiaries.java]: Added it.

File: rest-api/src/main/java/org/madb/api/config/DataInitializer.java [hunk 6]
```diff
@@ -47,31 +49,35 @@
         @Autowired
```
Comment (raftAtGit) [inline rest-api/src/main/java/org/madb/api/config/DataInitializer.java]: There are still lots of indentation errors :/

File: rest-api/src/main/java/org/madb/api/controller/PartnershipController.java [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/v1")
+@Validated
+
+public class PartnershipController {
+    @Autowired
+	private ProjectRepository projectRepository;
```
Comment (raftAtGit) [inline rest-api/src/main/java/org/madb/api/controller/PartnershipController.java]: indentation.. check the whole thing please

**Comment by raftAtGit** [comment]
there are conflicts. merge latest master branch to your branch please

File: rest-api/src/main/java/org/madb/api/config/DataInitializer.java (hunk 6 above)
Comment (HiradEmami) [reply rest-api/src/main/java/org/madb/api/config/DataInitializer.java]: I'm using Netbeans, didnt notice any indentation error.

Comment (raftAtGit) [reply rest-api/src/main/java/org/madb/api/config/DataInitializer.java]: it's visible here on GitHub or in Eclipse or in command line.
possibly related to your indentation settings in NetBeans

File: rest-api/src/main/java/org/madb/api/controller/PartnershipController.java (hunk 7 above)
Comment (HiradEmami) [reply rest-api/src/main/java/org/madb/api/controller/PartnershipController.java]: I fixed the extra tab here.

File: rest-api/src/main/java/org/madb/api/model/Project.java [hunk 8]
```diff
@@ -65,6 +65,11 @@
 	@Size(max = 100)
 	@Column(name = "OBSERVATIONS")
 	private String observations;
+        
+        @NotNull
```
Comment (raftAtGit) [inline rest-api/src/main/java/org/madb/api/model/Project.java]: almost ;)

**Pull request merged**